Update compile and instructions

leagris opened the pull request:
Fiabilize and make compile.sh use POSIX grammar.

Update build instructions to docker buildx as build is deprecated
and soon removed.

leagris committed:
Update compile and instructions

Fiabilize and make compile.sh use POSIX grammar.

Update build instructions to docker buildx as build is deprecated
and soon removed.

leagris committed:
minor markdown syntax fixes

robertaboukhalil commented on compile.sh:
```diff
@@ -1,20 +1,23 @@
-[[ ! $(command -v emcc --version) ]] && \
-  echo "You need Emscripten to compile jq to WebAssembly. See the setup instructions in the README file." && \
-  exit
+if ! command -v emcc > /dev/null 2>&1; then
+  printf '%s\n' \
+    'You need Emscripten to compile jq to WebAssembly.' \
+    'See the setup instructions in the README file.'
+  exit 2
+fi >&2
```
Can you undo lines 6-11? Not a fan of printf instead of echo 🙂

leagris replied:
The reason to not use `echo` is because this command behaves differently in different shells.
There is a good detailed explanation here https://www.in-ulm.de/~mascheck/various/echo+printf/

robertaboukhalil commented on compile.sh:
```diff
@@ -29,5 +32,5 @@ emmake make \
 
-mkdir -p ../build/
-mv jq.{js,wasm} ../build/
+mkdir -p -- ../build/
+mv -- jq.js jq.wasm ../build/
```
Can you undo these 2 lines? The `--` is confusing

leagris replied:
These are not needed in this specific case with constant paths indeed.

leagris committed:
Requested changes

https://github.com/robertaboukhalil/jqkungfu/pull/11#discussion_r1401486594
https://github.com/robertaboukhalil/jqkungfu/pull/11#discussion_r1401486806
